diff_drive_controller/package.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0"?>
+<package format="3">
+  <name>diff_drive_controller</name>
+  <version>0.0.1</version>
+  <description>Controller for executing joint-space trajectories on a group of joints</description>
+  <maintainer email="allabana@amazon.com">Anas Abou Allaban</maintainer>
+
+  <license>Apache License 2.0</license>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <build_depend>controller_interface</build_depend>
+  <build_depend>hardware_interface</build_depend>
+  <build_depend>pluginlib</build_depend>
+  <build_depend>rclcpp</build_depend>
+  <build_depend>rclcpp_lifecycle</build_depend>
+  <build_depend>rcutils</build_depend>
+  <build_depend>realtime_tools</build_depend>
+  <build_depend>geometry_msgs</build_depend>
+  <build_depend>nav_msgs</build_depend>
+  <build_depend>tf2_msgs</build_depend>
+  <build_depend>boost</build_depend>
```

Comment:
Could this be made more precise?

The `boost` key will drag in *all* of Boost ([here](https://github.com/ros/rosdistro/blob/fa9b8015feb7acebea9667f2dae443bb50da994b/rosdep/base.yaml#L275-L301)).

If only specific parts of Boost are needed, we should use the appropriate keys from [here](https://github.com/ros/rosdistro/blob/fa9b8015feb7acebea9667f2dae443bb50da994b/rosdep/base.yaml#L1624-L1826), which will significantly reduce the size of the Boost dependency.

Reply:
Thanks for pointing that out to me, I had no idea how to specify which boost library and couldn't find documentation for it

Reply:
Also it seems like `ament_target_dependencies` doesn't like cmake aliases, so I can't just pass in `Boost::boost` to get the header only stuff

Reply:
My "header only" reference was to the libraries themselves. The manifest is separate from the build script (ie: `CMakeLists.txt`).

Header-only libraries which are not included in any public headers don't need to be listed as `exec_depend`, nor as `build_export_depend`. So they would only appear as `build_depend`.